In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('nba.xlsx.csv')

In [3]:
df

Name            Team  Number Position  Age Height  Weight  \
0    Avery Bradley  Boston Celtics       0       PG   25    6-2     180   
1      Jae Crowder  Boston Celtics      99       SF   25    6-6     235   
2     John Holland  Boston Celtics      30       SG   27    6-5     205   
3      R.J. Hunter  Boston Celtics      28       SG   22    6-5     185   
4    Jonas Jerebko  Boston Celtics       8       PF   29   6-10     231   
..             ...             ...     ...      ...  ...    ...     ...   
452     Trey Lyles       Utah Jazz      41       PF   20   6-10     234   
453   Shelvin Mack       Utah Jazz       8       PG   26    6-3     203   
454      Raul Neto       Utah Jazz      25       PG   24    6-1     179   
455   Tibor Pleiss       Utah Jazz      21        C   26    7-3     256   
456    Jeff Withey       Utah Jazz      24        C   26    7-0     231   

               College     Salary  
0                Texas  7730337.0  
1            Marquette  6796117.0  
2    Boston University        NaN  
3        Georgia State  1148640.0  
4                  NaN  5000000.0  
..                 ...        ...  
452           Kentucky  2239800.0  
453             Butler  2433333.0  
454                NaN   900000.0  
455                NaN  2900000.0  
456             Kansas   947276.0  

[457 rows x 9 columns]

** Question 1 Select all rows excluding bottom 157 rows using head() function. What is the salary in the last row?

In [5]:
df.head(-157)

Name               Team  Number Position  Age Height  Weight  \
0    Avery Bradley     Boston Celtics       0       PG   25    6-2     180   
1      Jae Crowder     Boston Celtics      99       SF   25    6-6     235   
2     John Holland     Boston Celtics      30       SG   27    6-5     205   
3      R.J. Hunter     Boston Celtics      28       SG   22    6-5     185   
4    Jonas Jerebko     Boston Celtics       8       PF   29   6-10     231   
..             ...                ...     ...      ...  ...    ...     ...   
295  Kyle Anderson  San Antonio Spurs       1       SF   22    6-9     230   
296    Matt Bonner  San Antonio Spurs      15        C   36   6-10     235   
297     Boris Diaw  San Antonio Spurs      33        C   34    6-8     250   
298     Tim Duncan  San Antonio Spurs      21        C   40   6-11     250   
299  Manu Ginobili  San Antonio Spurs      20       SG   38    6-6     205   

               College     Salary  
0                Texas  7730337.0  
1            Marquette  6796117.0  
2    Boston University        NaN  
3        Georgia State  1148640.0  
4                  NaN  5000000.0  
..                 ...        ...  
295               UCLA  1142880.0  
296            Florida   947276.0  
297                NaN  7500000.0  
298        Wake Forest  5250000.0  
299                NaN  2814000.0  

[300 rows x 9 columns]

** Question 2 Select all rows excluding bottom 17 rows using head() function. What is the salary in the last row?

In [7]:
df.head(17)

Name            Team  Number Position  Age Height  Weight  \
0      Avery Bradley  Boston Celtics       0       PG   25    6-2     180   
1        Jae Crowder  Boston Celtics      99       SF   25    6-6     235   
2       John Holland  Boston Celtics      30       SG   27    6-5     205   
3        R.J. Hunter  Boston Celtics      28       SG   22    6-5     185   
4      Jonas Jerebko  Boston Celtics       8       PF   29   6-10     231   
5       Amir Johnson  Boston Celtics      90       PF   29    6-9     240   
6      Jordan Mickey  Boston Celtics      55       PF   21    6-8     235   
7       Kelly Olynyk  Boston Celtics      41        C   25    7-0     238   
8       Terry Rozier  Boston Celtics      12       PG   22    6-2     190   
9       Marcus Smart  Boston Celtics      36       PG   22    6-4     220   
10   Jared Sullinger  Boston Celtics       7        C   24    6-9     260   
11     Isaiah Thomas  Boston Celtics       4       PG   27    5-9     185   
12       Evan Turner  Boston Celtics      11       SG   27    6-7     220   
13       James Young  Boston Celtics      13       SG   20    6-6     215   
14      Tyler Zeller  Boston Celtics      44        C   26    7-0     253   
15  Bojan Bogdanovic   Brooklyn Nets      44       SG   27    6-8     216   
16      Markel Brown   Brooklyn Nets      22       SG   24    6-3     190   

              College      Salary  
0               Texas   7730337.0  
1           Marquette   6796117.0  
2   Boston University         NaN  
3       Georgia State   1148640.0  
4                 NaN   5000000.0  
5                 NaN  12000000.0  
6                 LSU   1170960.0  
7             Gonzaga   2165160.0  
8          Louisville   1824360.0  
9      Oklahoma State   3431040.0  
10         Ohio State   2569260.0  
11         Washington   6912869.0  
12         Ohio State   3425510.0  
13           Kentucky   1749840.0  
14     North Carolina   2616975.0  
15                NaN   3425510.0  
16     Oklahoma State    845059.0

** Question 3 What is the memory usage (memory_usage with default arguments) of NBA dataset columns?

Share the memory usage of column “Number”?

In [8]:
df.memory_usage()

Index        128
Name        3656
Team        3656
Number      3656
Position    3656
Age         3656
Height      3656
Weight      3656
College     3656
Salary      3656
dtype: int64

** Question 4 Share the memory usage of column “Name” we deeply introspect system-level memory consumption data?

In [9]:
df.memory_usage(deep = True)

Index         128
Name        32008
Team        33405
Number       3656
Position    26885
Age          3656
Height      27510
Weight       3656
College     27236
Salary       3656
dtype: int64

** Question 5

How many NA values are present in the column Salary in the NBA dataset?

In [13]:
df.isna().sum()

Name         0
Team         0
Number       0
Position     0
Age          0
Height       0
Weight       0
College     84
Salary      11
dtype: int64

** Question 6 Drop rows (25,72,63) from the NBA dataset. What is the average age of the players after dropping rows?

Note: Please do not change the original dataset while solving this assignment.


In [15]:
df.drop([25, 72, 63],axis =0)
df.Age.mean()

26.938730853391686

** Question 7 Rename column ‘Number’ to ‘Chest_Number’ and ‘Weight’ to ‘Weight(lbs)’ ’in the NBA dataset.
Use inplace argument to reflect in the original dataset

In [16]:
df.rename(columns={"Number":"Chest_Number",
                  "Weight":"Weight(lbs)"})

Name            Team  Chest_Number Position  Age Height  \
0    Avery Bradley  Boston Celtics             0       PG   25    6-2   
1      Jae Crowder  Boston Celtics            99       SF   25    6-6   
2     John Holland  Boston Celtics            30       SG   27    6-5   
3      R.J. Hunter  Boston Celtics            28       SG   22    6-5   
4    Jonas Jerebko  Boston Celtics             8       PF   29   6-10   
..             ...             ...           ...      ...  ...    ...   
452     Trey Lyles       Utah Jazz            41       PF   20   6-10   
453   Shelvin Mack       Utah Jazz             8       PG   26    6-3   
454      Raul Neto       Utah Jazz            25       PG   24    6-1   
455   Tibor Pleiss       Utah Jazz            21        C   26    7-3   
456    Jeff Withey       Utah Jazz            24        C   26    7-0   

     Weight(lbs)            College     Salary  
0            180              Texas  7730337.0  
1            235          Marquette  6796117.0  
2            205  Boston University        NaN  
3            185      Georgia State  1148640.0  
4            231                NaN  5000000.0  
..           ...                ...        ...  
452          234           Kentucky  2239800.0  
453          203             Butler  2433333.0  
454          179                NaN   900000.0  
455          256                NaN  2900000.0  
456          231             Kansas   947276.0  

[457 rows x 9 columns]

** Question 8 Filter the data with more than 35 years of age. Count the number of names and share it in the answer

In [18]:
df[(df["Age"] >35)].value_counts("Name").count()

23

** Question 9 Filter the data with more than 35 years of age.
Find the number of records for the “PF” position.

In [24]:
df[(df["Age"] >35) & (df["Position"]=="PF")]

Name                    Team  Number Position  Age Height  \
46      Elton Brand      Philadelphia 76ers      42       PF   37    6-9   
72       Luis Scola         Toronto Raptors       4       PF   36    6-9   
236   Dirk Nowitzki        Dallas Mavericks      41       PF   37    7-0   
259  Chris Andersen       Memphis Grizzlies       7       PF   37   6-10   
343   Udonis Haslem              Miami Heat      40       PF   36    6-8   
400   Kevin Garnett  Minnesota Timberwolves      21       PF   40   6-11   

     Weight        College     Salary  
46      254           Duke        NaN  
72      240            NaN  2900000.0  
236     245            NaN  8333334.0  
259     245  Blinn College  5000000.0  
343     235        Florida  2854940.0  
400     240            NaN  8500000.0

** Question 10 Find the number of records for the “PG” position.

In [25]:
df[(df["Age"] >35) & (df["Position"]=="PG")]

Name                  Team  Number Position  Age Height  \
102  Pablo Prigioni  Los Angeles Clippers       9       PG   39    6-3   
183     Steve Blake       Detroit Pistons      22       PG   36    6-3   
304    Andre Miller     San Antonio Spurs      24       PG   40    6-3   

     Weight   College     Salary  
102     185       NaN   947726.0  
183     172  Maryland  2170465.0  
304     200      Utah   250750.0

** Question 11 Which team has the highest number of youngest players (Age less than or equal to 25) playing in the position of “PF” and “SF”.

In [ ]:
df[(df["Age"] <=25) &(df["Position"].isin(["PF","SF"]))]["Team"].value_counts()

** Question 12 How many team are there as per above question?

In [27]:
df[(df["Age"] <=25) &(df["Position"].isin(["PF","SF"]))]["Team"].value_counts().count()

29

** Question 13 As per the filter from previous question, find out the highest salary. Submit it in the answer.

In [28]:
df[(df["Age"] <=25) &(df["Position"].isin(["PF","SF"]))]["Salary"].max()

16407500.0

** Question 14 Enter the name of the player who has highest salary.

In [32]:
df[(df["Age"] <=25) &(df["Position"].isin(["PF","SF","Name"])) &df["Salary"].isin([16407500.0])]

Name               Team  Number Position  Age Height  Weight  \
301  Kawhi Leonard  San Antonio Spurs       2       SF   24    6-7     230   

             College      Salary  
301  San Diego State  16407500.0

** Question 15 How many players are between 25 to 35 of age?

In [35]:
df[df["Age"].between(25,35)]

Name            Team  Number Position  Age Height  Weight  \
0    Avery Bradley  Boston Celtics       0       PG   25    6-2     180   
1      Jae Crowder  Boston Celtics      99       SF   25    6-6     235   
2     John Holland  Boston Celtics      30       SG   27    6-5     205   
4    Jonas Jerebko  Boston Celtics       8       PF   29   6-10     231   
5     Amir Johnson  Boston Celtics      90       PF   29    6-9     240   
..             ...             ...     ...      ...  ...    ...     ...   
450     Joe Ingles       Utah Jazz       2       SF   28    6-8     226   
451  Chris Johnson       Utah Jazz      23       SF   26    6-6     206   
453   Shelvin Mack       Utah Jazz       8       PG   26    6-3     203   
455   Tibor Pleiss       Utah Jazz      21        C   26    7-3     256   
456    Jeff Withey       Utah Jazz      24        C   26    7-0     231   

               College      Salary  
0                Texas   7730337.0  
1            Marquette   6796117.0  
2    Boston University         NaN  
4                  NaN   5000000.0  
5                  NaN  12000000.0  
..                 ...         ...  
450                NaN   2050000.0  
451             Dayton    981348.0  
453             Butler   2433333.0  
455                NaN   2900000.0  
456             Kansas    947276.0  

[280 rows x 9 columns]

** Question 16 What is the sum of the salaries of all the players with chest number 41?

In [5]:
df.loc[df["Number"]==41,"Salary"].sum()

22264701.0

** Question 17 What is the average salary received by players who are 23 or younger in the team ‘Chicago Bulls’?

In [31]:
df[(df["Age"] <= 23) & (df['Team'] == "Chicago Bulls")]["Salary"].mean()

958126.5

** Question 18 What is the average age of players from rows 201 to 250?

In [7]:
df.loc[201:249]["Age"].mean()

26.714285714285715

** Question 19 What is the average age of players in the team Boston Celtics?
Also, round it off to 2 decimals

In [8]:
round(df[df["Team"] == "Boston Celtics"]["Age"].mean(),2)


24.73

** Consider the following DataFrame. 


In [8]:
students = [['Jack', 34, 'Sydney', 'Australia',85.96],

            ['Riti', 30, 'Delhi', 'India',95.20],

            ['Vansh', 31, 'Delhi', 'India',85.25],

            ['Nanyu', 32, 'Tokyo', 'Japan',74.21],

            ['Maychan', 16, 'New York', 'US',99.63],

            ['Mike', 17, 'las vegas', 'US',47.28]]

 

df_s = pd.DataFrame(students,

                      columns=['Name', 'Age', 'City', 'Country','Agg_Marks'],

                           index=['ID01', 'ID02', 'ID03', 'ID04', 'ID05', 'ID06'])

In [9]:
df_s

Name  Age       City    Country  Agg_Marks
ID01     Jack   34     Sydney  Australia      85.96
ID02     Riti   30      Delhi      India      95.20
ID03    Vansh   31      Delhi      India      85.25
ID04    Nanyu   32      Tokyo      Japan      74.21
ID05  Maychan   16   New York         US      99.63
ID06     Mike   17  las vegas         US      47.28

In [10]:
df_s.iloc[1:3, 1:3]

Age   City
ID02   30  Delhi
ID03   31  Delhi

In [11]:
df_s.iloc[[1,3], [1,3]]

Age Country
ID02   30   India
ID04   32   Japan

In [12]:
df_s.iloc[0:10, [1,3]]

Age    Country
ID01   34  Australia
ID02   30      India
ID03   31      India
ID04   32      Japan
ID05   16         US
ID06   17         US

In [13]:
df_s.iloc[:, [1,3]]

Age    Country
ID01   34  Australia
ID02   30      India
ID03   31      India
ID04   32      Japan
ID05   16         US
ID06   17         US

In [14]:
df_s.loc[:, ["Age", "Country"]]

Age    Country
ID01   34  Australia
ID02   30      India
ID03   31      India
ID04   32      Japan
ID05   16         US
ID06   17         US

In [15]:
df_s.loc[['ID01', 'ID02', 'ID03', 'ID04', 'ID05', 'ID06'], ["Age", "Country"]]

Age    Country
ID01   34  Australia
ID02   30      India
ID03   31      India
ID04   32      Japan
ID05   16         US
ID06   17         US

In [16]:
df_s.loc[:]

Name  Age       City    Country  Agg_Marks
ID01     Jack   34     Sydney  Australia      85.96
ID02     Riti   30      Delhi      India      95.20
ID03    Vansh   31      Delhi      India      85.25
ID04    Nanyu   32      Tokyo      Japan      74.21
ID05  Maychan   16   New York         US      99.63
ID06     Mike   17  las vegas         US      47.28

In [17]:
df_s.loc[:, :]

Name  Age       City    Country  Agg_Marks
ID01     Jack   34     Sydney  Australia      85.96
ID02     Riti   30      Delhi      India      95.20
ID03    Vansh   31      Delhi      India      85.25
ID04    Nanyu   32      Tokyo      Japan      74.21
ID05  Maychan   16   New York         US      99.63
ID06     Mike   17  las vegas         US      47.28

In [18]:
df_s.iloc[:]

Name  Age       City    Country  Agg_Marks
ID01     Jack   34     Sydney  Australia      85.96
ID02     Riti   30      Delhi      India      95.20
ID03    Vansh   31      Delhi      India      85.25
ID04    Nanyu   32      Tokyo      Japan      74.21
ID05  Maychan   16   New York         US      99.63
ID06     Mike   17  las vegas         US      47.28

In [19]:
df_s.iloc[:, :]

Name  Age       City    Country  Agg_Marks
ID01     Jack   34     Sydney  Australia      85.96
ID02     Riti   30      Delhi      India      95.20
ID03    Vansh   31      Delhi      India      85.25
ID04    Nanyu   32      Tokyo      Japan      74.21
ID05  Maychan   16   New York         US      99.63
ID06     Mike   17  las vegas         US      47.28

In [20]:
df_s[:]

Name  Age       City    Country  Agg_Marks
ID01     Jack   34     Sydney  Australia      85.96
ID02     Riti   30      Delhi      India      95.20
ID03    Vansh   31      Delhi      India      85.25
ID04    Nanyu   32      Tokyo      Japan      74.21
ID05  Maychan   16   New York         US      99.63
ID06     Mike   17  las vegas         US      47.28

In [21]:
df_s

Name  Age       City    Country  Agg_Marks
ID01     Jack   34     Sydney  Australia      85.96
ID02     Riti   30      Delhi      India      95.20
ID03    Vansh   31      Delhi      India      85.25
ID04    Nanyu   32      Tokyo      Japan      74.21
ID05  Maychan   16   New York         US      99.63
ID06     Mike   17  las vegas         US      47.28

*** Create a new column where height is in centimeters. Name the column as "Height (cm)"

Instruction: Height is given in feet and inches such as 6-2; 6 feet and 2 inches


Open Hint: 

Google the logic of converting height from feet - inches into centimeters
Use the string function to split the string
Understand and use the map function in Python
Create a dedicated function 
Use Pandas apply function with lambda to convert for all the data in the “Height” column

Note: Please make sure to get your code reviewed by mentors.

In [73]:
df[['Feet', 'Inches']] = df['Height'].str.split('-', expand=True)

In [74]:
df[['Feet', 'Inches']] = df[['Feet', 'Inches']].applymap(int)

In [75]:
def convert_height(feet, inches):
    total_inches = feet * 12 + inches
    return round(total_inches * 2.54, 2)


In [76]:
df['Height_cm'] = df.apply(lambda x: convert_height(x['Feet'], x['Inches']), axis=1)

In [77]:
df

Name            Team  Number Position  Age Height  Weight  \
0    Avery Bradley  Boston Celtics       0       PG   25    6-2     180   
1      Jae Crowder  Boston Celtics      99       SF   25    6-6     235   
2     John Holland  Boston Celtics      30       SG   27    6-5     205   
3      R.J. Hunter  Boston Celtics      28       SG   22    6-5     185   
4    Jonas Jerebko  Boston Celtics       8       PF   29   6-10     231   
..             ...             ...     ...      ...  ...    ...     ...   
452     Trey Lyles       Utah Jazz      41       PF   20   6-10     234   
453   Shelvin Mack       Utah Jazz       8       PG   26    6-3     203   
454      Raul Neto       Utah Jazz      25       PG   24    6-1     179   
455   Tibor Pleiss       Utah Jazz      21        C   26    7-3     256   
456    Jeff Withey       Utah Jazz      24        C   26    7-0     231   

               College     Salary  Feet  Inches  Height_cm  
0                Texas  7730337.0     6       2     187.96  
1            Marquette  6796117.0     6       6     198.12  
2    Boston University        NaN     6       5     195.58  
3        Georgia State  1148640.0     6       5     195.58  
4                  NaN  5000000.0     6      10     208.28  
..                 ...        ...   ...     ...        ...  
452           Kentucky  2239800.0     6      10     208.28  
453             Butler  2433333.0     6       3     190.50  
454                NaN   900000.0     6       1     185.42  
455                NaN  2900000.0     7       3     220.98  
456             Kansas   947276.0     7       0     213.36  

[457 rows x 12 columns]

** Question 20 What is the height in cms for 112th record (111th index)?

In [78]:
df.loc[111,"Height_cm"]

218.44

In [79]:
df

Name            Team  Number Position  Age Height  Weight  \
0    Avery Bradley  Boston Celtics       0       PG   25    6-2     180   
1      Jae Crowder  Boston Celtics      99       SF   25    6-6     235   
2     John Holland  Boston Celtics      30       SG   27    6-5     205   
3      R.J. Hunter  Boston Celtics      28       SG   22    6-5     185   
4    Jonas Jerebko  Boston Celtics       8       PF   29   6-10     231   
..             ...             ...     ...      ...  ...    ...     ...   
452     Trey Lyles       Utah Jazz      41       PF   20   6-10     234   
453   Shelvin Mack       Utah Jazz       8       PG   26    6-3     203   
454      Raul Neto       Utah Jazz      25       PG   24    6-1     179   
455   Tibor Pleiss       Utah Jazz      21        C   26    7-3     256   
456    Jeff Withey       Utah Jazz      24        C   26    7-0     231   

               College     Salary  Feet  Inches  Height_cm  
0                Texas  7730337.0     6       2     187.96  
1            Marquette  6796117.0     6       6     198.12  
2    Boston University        NaN     6       5     195.58  
3        Georgia State  1148640.0     6       5     195.58  
4                  NaN  5000000.0     6      10     208.28  
..                 ...        ...   ...     ...        ...  
452           Kentucky  2239800.0     6      10     208.28  
453             Butler  2433333.0     6       3     190.50  
454                NaN   900000.0     6       1     185.42  
455                NaN  2900000.0     7       3     220.98  
456             Kansas   947276.0     7       0     213.36  

[457 rows x 12 columns]

*** Currently “Weight” in lbs. Create a new column where “weight” is in kg. Name the column "Weight(kg)" 

Note: Please make sure to get your code reviewed by mentors.

In [80]:
df["Weight(kg)"]= round(df["Weight"]*0.453592,2)

In [81]:
df

Name            Team  Number Position  Age Height  Weight  \
0    Avery Bradley  Boston Celtics       0       PG   25    6-2     180   
1      Jae Crowder  Boston Celtics      99       SF   25    6-6     235   
2     John Holland  Boston Celtics      30       SG   27    6-5     205   
3      R.J. Hunter  Boston Celtics      28       SG   22    6-5     185   
4    Jonas Jerebko  Boston Celtics       8       PF   29   6-10     231   
..             ...             ...     ...      ...  ...    ...     ...   
452     Trey Lyles       Utah Jazz      41       PF   20   6-10     234   
453   Shelvin Mack       Utah Jazz       8       PG   26    6-3     203   
454      Raul Neto       Utah Jazz      25       PG   24    6-1     179   
455   Tibor Pleiss       Utah Jazz      21        C   26    7-3     256   
456    Jeff Withey       Utah Jazz      24        C   26    7-0     231   

               College     Salary  Feet  Inches  Height_cm  Weight(kg)  
0                Texas  7730337.0     6       2     187.96       81.65  
1            Marquette  6796117.0     6       6     198.12      106.59  
2    Boston University        NaN     6       5     195.58       92.99  
3        Georgia State  1148640.0     6       5     195.58       83.91  
4                  NaN  5000000.0     6      10     208.28      104.78  
..                 ...        ...   ...     ...        ...         ...  
452           Kentucky  2239800.0     6      10     208.28      106.14  
453             Butler  2433333.0     6       3     190.50       92.08  
454                NaN   900000.0     6       1     185.42       81.19  
455                NaN  2900000.0     7       3     220.98      116.12  
456             Kansas   947276.0     7       0     213.36      104.78  

[457 rows x 13 columns]

** Question 21 What is the weight in kg for 112th record (111th index)?

In [64]:
df.loc[111,"Weight(kg)"]

122.47

*** Create a new column to calculate “BMI” for each player.


Open Hint: 

1.Google the formula of BMI.

2.This column can be created without using the Pandas apply function with lambda. Try it.

Note: Please make sure to get your code reviewed by mentors.

In [93]:
df["BMI"] = round(df["Weight(kg)"]/((df["Height_cm"] / 100)**2),2)

** Question 22 What is the BMI for 112th record (111th index)?

In [87]:
df.loc[111,"BMI"]

25.67

** Question 23 What is the maximum BMI?

In [96]:
df["BMI"].max()

31.33

** Question 24 What is the minimum BMI?

In [97]:
df["BMI"].min()

19.93

In [103]:
import numpy as np

*** Create a new column with the name “BMI Evaluation”. Assign the BMI status based on the following condition,


BMI <= 18.4 - "Underweight"

BMI <= 24.9 - "Healthy"

BMI <= 29.9 - "Over Weight"

BMI <= 34.9 - "Severely Over Weight"

BMI <= 39.9 - "Obese"
39.9 - "Severely Obese."

In [104]:
conditions = [
    (df['BMI'] <= 18.4),
    (df['BMI'] <= 24.9),
    (df['BMI'] <= 29.9),
    (df['BMI'] <= 34.9),
    (df['BMI'] <= 39.9),
    (df['BMI'] > 39.9)
]

In [105]:
bmi_categories = [
    'Underweight',
    'Healthy',
    'Over Weight',
    'Severely Over Weight',
    'Obese',
    'Severely Obese'
]

In [106]:
df["BMI Evaluation"] = np.select(conditions, bmi_categories)

In [107]:
df

Name            Team  Number Position  Age Height  Weight  \
0    Avery Bradley  Boston Celtics       0       PG   25    6-2     180   
1      Jae Crowder  Boston Celtics      99       SF   25    6-6     235   
2     John Holland  Boston Celtics      30       SG   27    6-5     205   
3      R.J. Hunter  Boston Celtics      28       SG   22    6-5     185   
4    Jonas Jerebko  Boston Celtics       8       PF   29   6-10     231   
..             ...             ...     ...      ...  ...    ...     ...   
452     Trey Lyles       Utah Jazz      41       PF   20   6-10     234   
453   Shelvin Mack       Utah Jazz       8       PG   26    6-3     203   
454      Raul Neto       Utah Jazz      25       PG   24    6-1     179   
455   Tibor Pleiss       Utah Jazz      21        C   26    7-3     256   
456    Jeff Withey       Utah Jazz      24        C   26    7-0     231   

               College     Salary  Feet  Inches  Height_cm  Weight(kg)    BMI  \
0                Texas  7730337.0     6       2     187.96       81.65  23.11   
1            Marquette  6796117.0     6       6     198.12      106.59  27.16   
2    Boston University        NaN     6       5     195.58       92.99  24.31   
3        Georgia State  1148640.0     6       5     195.58       83.91  21.94   
4                  NaN  5000000.0     6      10     208.28      104.78  24.15   
..                 ...        ...   ...     ...        ...         ...    ...   
452           Kentucky  2239800.0     6      10     208.28      106.14  24.47   
453             Butler  2433333.0     6       3     190.50       92.08  25.37   
454                NaN   900000.0     6       1     185.42       81.19  23.62   
455                NaN  2900000.0     7       3     220.98      116.12  23.78   
456             Kansas   947276.0     7       0     213.36      104.78  23.02   

     Height(m) BMI Evaluation  
0       1.8796        Healthy  
1       1.9812    Over Weight  
2       1.9558        Healthy  
3       1.9558        Healthy  
4       2.0828        Healthy  
..         ...            ...  
452     2.0828        Healthy  
453     1.9050    Over Weight  
454     1.8542        Healthy  
455     2.2098        Healthy  
456     2.1336        Healthy  

[457 rows x 16 columns]

** Question 25 Count the occurrence of each BMI status and answer the following?
How many players have BMI as “Healthy”?

In [108]:
df[df["BMI Evaluation"] == "Healthy"].shape[0]

247

** Question 26 How many players have BMI as “Over Weight”?

In [110]:
df[df["BMI Evaluation"]=="Over Weight"].shape[0]

208

** Question 27 How many players have BMI as “Obese”?

In [112]:
df[df["BMI Evaluation"]=="Obese"].shape[0]

0

** Question 28 Compute the total amount paid by each team to the players.
What is the amount paid by Chicago Bulls to the players?

In [113]:
df[df["Team"] == "Chicago Bulls"]["Salary"].sum()

86783378.0

** Question 29 Compute the average age of each team by their playing position. What is the average age of the Utah Jazz team players playing at the ‘PG’ position?

In [5]:
df[(df["Team"] == "Utah Jazz") & (df["Position"] == "PG")]["Age"].mean()

23.25

** Question 30 Compute each team's average, maximum, and minimum age by their playing position. What is the average, maximum, and minimum age of the Atlanta Hawks players playing at the ‘PG’ position?

Enter average age:

In [6]:
df[(df["Team"]=="Atlanta Hawks") & (df["Position"]=="PG")]["Age"].agg(["mean","min","max"])

mean    24.5
min     22.0
max     27.0
Name: Age, dtype: float64

** Question 31 Compute each team's average age and total salary by their playing position. Sort the grouped results based on salary.
Enter the team name with the highest salary.

In [31]:
 df.groupby(['Team', 'Position']).agg({'Age': 'mean', 'Salary': 'sum'}).sort_values('Salary', ascending=False)

Age      Salary
Team               Position                   
Los Angeles Lakers SF        31.75  31866445.0
Miami Heat         PF        32.50  31538671.0
Houston Rockets    SG        29.25  28122883.0
Phoenix Suns       PG        26.75  28002998.0
Denver Nuggets     SF        27.00  27982771.0
...                            ...         ...
                   SG        25.75   2790245.0
Philadelphia 76ers SF        23.50   1845059.0
Sacramento Kings   PF        24.00   1370152.0
Miami Heat         C         26.00    981348.0
Orlando Magic      SF        23.00    845059.0

[149 rows x 2 columns]

*** Compute the mean salary of the team using group by and create dataframe df_1.

Join the tables df_1 and df to add the mean salary of each team in every row [Similar to Avg Over Partition By Windows Function in SQL]

In [26]:
df_1 = df.merge(df.groupby("Team")["Salary"].mean(),on="Team",suffixes=("_player","_team"))
df_1

Name            Team  Number Position  Age Height  Weight  \
0    Avery Bradley  Boston Celtics       0       PG   25    6-2     180   
1      Jae Crowder  Boston Celtics      99       SF   25    6-6     235   
2     John Holland  Boston Celtics      30       SG   27    6-5     205   
3      R.J. Hunter  Boston Celtics      28       SG   22    6-5     185   
4    Jonas Jerebko  Boston Celtics       8       PF   29   6-10     231   
..             ...             ...     ...      ...  ...    ...     ...   
452     Trey Lyles       Utah Jazz      41       PF   20   6-10     234   
453   Shelvin Mack       Utah Jazz       8       PG   26    6-3     203   
454      Raul Neto       Utah Jazz      25       PG   24    6-1     179   
455   Tibor Pleiss       Utah Jazz      21        C   26    7-3     256   
456    Jeff Withey       Utah Jazz      24        C   26    7-0     231   

               College  Salary_player   Salary_team  
0                Texas      7730337.0  4.181505e+06  
1            Marquette      6796117.0  4.181505e+06  
2    Boston University            NaN  4.181505e+06  
3        Georgia State      1148640.0  4.181505e+06  
4                  NaN      5000000.0  4.181505e+06  
..                 ...            ...           ...  
452           Kentucky      2239800.0  4.204006e+06  
453             Butler      2433333.0  4.204006e+06  
454                NaN       900000.0  4.204006e+06  
455                NaN      2900000.0  4.204006e+06  
456             Kansas       947276.0  4.204006e+06  

[457 rows x 10 columns]

** Question 32 How many players in the whole dataset receive a salary more than the average salary of their team?

In [27]:
df_1[df_1["Salary_player"]>df_1["Salary_team"]].shape[0]

148

*** Saving the data in CSV or Excel

In [28]:
df.to_csv('nba.xlsx.csv',index = True)